### ISSUE TITLE
Cache resolved used mappers and prototype methods

### ISSUE THREAD

**Issue by gunnarmorling** [mapstruct/mapstruct#473]
Currently we resolve each mapper referenced via `uses()` and prototype methods given in config classes repeatedly for each usage. This should be cached by used mapper / config class upon first resolution.

**Comment by agudian** [comment]
Good idea! We need to take care to root the lifetime of the cache somehow to the lifetime of the processor instance, though. Otherwise we might get funny effects during incremental compilation in Eclipse.
